src/TestExplorer/TestParsers/SwiftTestingOutputParser.ts

```diff
@@ -155,31 +176,131 @@ export class SwiftTestingOutputParser {
+    // with Codable. If they aren't, their id appears as `argumentIDs: nil`, and we
+    // fall back to using the testCase display name as the test case ID. This isn't
+    // ideal because its possible to have multiple test cases with the same display name,
+    // but until we have a better solution for identifying test cases it will have to do.
+    // SEE: rdar://119522099.
+    private createTestCaseId(testCase: TestCase): string {
+        return testCase.id === "argumentIDs: nil" ? testCase.displayName : testCase.id;
+    }
+
+    private parameterizedFunctionTestCaseToTestClass(
+        testCase: TestCase,
+        location: vscode.Location,
+        index: number
+    ): TestClass {
+        return {
+            id: this.createTestCaseId(testCase),
+            label: testCase.displayName,
+            tags: [],
+            children: [],
+            style: "swift-testing",
+            location: location,
+            disabled: true,
+            sortText: `${index}`.padStart(8, "0"),
+        };
+    }
+
+    private buildTestCaseMapForParameterizedTest(record: TestRecord) {
+        const map = new Map<string, TestCase>();
+        (record.payload._testCases ?? []).forEach(testCase => {
+            map.set(this.createTestCaseId(testCase), testCase);
+        });
+        this.testCaseMap.set(record.payload.id, map);
+    }
+
     private parse(item: SwiftTestEvent, runState: ITestRunState) {
-        if (item.kind === "event") {
-            if (item.payload.kind === "testCaseStarted" || item.payload.kind === "testStarted") {
+        if (
+            item.kind === "test" &&
+            item.payload.kind === "function" &&
+            item.payload.isParameterized &&
+            item.payload._testCases
+        ) {
+            // Store a map of [Test ID, [Test Case ID, TestCase]] so we can quickly
+            // map an event.payload.testID back to a test case.
+            this.buildTestCaseMapForParameterizedTest(item);
+
+            const testName = this.testName(item.payload.id);
+            const testIndex = runState.getTestItemIndex(testName, undefined);
+            // If a test has test cases it is paramterized and we need to notify
+            // the caller that the TestClass should be added to the vscode.TestRun
+            // before it starts.
+            item.payload._testCases
+                .map((testCase, index) =>
+                    this.parameterizedFunctionTestCaseToTestClass(
+                        testCase,
+                        sourceLocationToVSCodeLocation(item.payload.sourceLocation),
+                        index
+                    )
+                )
+                .flatMap(testClass => (testClass ? [testClass] : []))
+                .forEach(testClass => this.addParameterizedTestCase(testClass, testIndex));
+        } else if (item.kind === "event") {
+            if (item.payload.kind === "runStarted") {
```

Comment:
suggestion: again maybe some info about where these payload kinds come from and what they mean